ARM ID validation

matthchr opened the pull request:
Related to #765

matthchr committed:
ARM ID validation

raych1 reviewed (changes requested):
Can you add some unit tests for arm-id validation at
https://github.com/Azure/oav/blob/develop/test/modelValidatorTests.ts

matthchr committed:
Unit tests

matthchr commented:
I've added a unit test @raych1. Let me know if you feel there should be more. I tried to follow the other format tests that I saw (`byte`).

raych1 committed:
Restrict arm-id validation to examples and arm tra

raych1 committed:
Merge branch 'develop' into feature/arm-id-validation

raych1 committed:
Refactor missing_resource_id rule

Ignore error when response body is empty in example validation;
Add isArmCall flag in semantic validation.

raych1 committed:
Updated snapshot of example generation test

Test failed because the location is set as required property in resource model so response body only has location property and doesn't have id property.

raych1 committed:
Merge branch 'develop' into feature/arm-id-validation

raych1 committed:
Updated snapshot for example validation test

raych1 committed:
Merge branch 'develop' into feature/arm-id-validation

raych1 committed:
Delete examplegen snapshot

raych1 committed:
re-add example gen snapshot

raych1 committed:
add swagger changes

raych1 committed:
add example files used by test

raych1 committed:
added examples used by test

raych1 committed:
Added all exmaples generated by auto gen

jianyexi commented on lib/swaggerValidator/modelValidator.ts:
```diff
@@ -573,6 +573,16 @@ export class SwaggerExampleValidator {
+
+        if (
+          (err.code as any) === "MISSING_RESOURCE_ID" &&
+          exampleContent.responses[`${statusCode}`].body &&
```
```suggestion
          exampleContent.responses[statusCode].body &&
```

jianyexi commented on lib/swaggerValidator/modelValidator.ts:
```diff
@@ -573,6 +573,16 @@ export class SwaggerExampleValidator {
+        if (
+          (err.code as any) === "MISSING_RESOURCE_ID" &&
+          exampleContent.responses[`${statusCode}`].body &&
+          Object.keys(exampleContent.responses[`${statusCode}`].body).length === 0
```
```suggestion
          Object.keys(exampleContent.responses[statusCode].body).length === 0
```

raych1 committed:
update index value in model validation

raych1 committed:
Merge branch 'develop' into feature/arm-id-validation
